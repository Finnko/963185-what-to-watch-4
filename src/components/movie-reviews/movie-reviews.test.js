import React from 'react';
import renderer from 'react-test-renderer';
import MovieReviews from './movie-reviews.jsx';

const mock = [
  {
    id: `1`,
    comment: `Good comment!`,
    rating: 8.9,
    user: `Ozzy Osbourne`,
    date: new Date(1583591483969)
  },
  {
    id: `2`,
    comment: `Neutral comment!`,
    rating: 5.9,
    user: `James Hatfield`,
    date: new Date(1593591555969)
  },
  {
    id: `3`,
    comment: `Bad comment!`,
    rating: 1.9,
    user: `Axl Rose`,
    date: new Date(1783591499969)
  },
];

describe(`MovieReviews component render correctly`, () => {
  it(`Should MovieReviews component render correctly`, () => {
    const tree = renderer
      .create(
          <MovieReviews
            reviews={mock}
          />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
