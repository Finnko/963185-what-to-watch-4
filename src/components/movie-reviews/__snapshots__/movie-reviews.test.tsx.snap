// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieReviews component render correctly Should MovieReviews component render correctly  1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Good comment!
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Ozzy Osbourne
          </cite>
          <time
            className="review__date"
            dateTime="2020-07-09"
          >
            July 09, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        8.9
      </div>
    </div>
  </div>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Good comment!
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Axl Rose
          </cite>
          <time
            className="review__date"
            dateTime="2020-07-09"
          >
            July 09, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        8.9
      </div>
    </div>
  </div>
</div>
`;

exports[`MovieReviews component render correctly Should MovieReviews component render correctly on loading 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="loader-wrapper"
    style={
      Object {
        "position": "absolute",
      }
    }
  >
    <div
      className="loader medium"
    />
  </div>
</div>
`;

exports[`MovieReviews component render correctly Should MovieReviews component render correctly when error occurred 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    Something went wrong, please try again.
  </div>
</div>
`;

exports[`MovieReviews component render correctly Should MovieReviews component render correctly without comments 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    There are no comments for this movie. Add the first one!
  </div>
</div>
`;
